src/api/spec/bootstrap/features/webui/attributes_spec.rb

```diff
@@ -0,0 +1,53 @@
+require 'browser_helper'
+
+RSpec.feature 'Bootstrap_Attributes', type: :feature, js: true do
+  let!(:user) { create(:confirmed_user) }
+  # AttribTypes are part of the seeds, so we can reuse them
+  let!(:attribute_type) { AttribType.find_by(name: 'ImageTemplates') }
+  let!(:attribute) { create(:attrib, project_id: user.home_project.id) }
+
+  context 'for a project' do
+    context 'without permissions' do
+      let!(:other_user) { create(:confirmed_user) }
+
+      scenario 'it is not possible to add an attribute, the link is not shown' do
+        login other_user
+
+        visit index_attribs_path(project: user.home_project_name)
+        expect(page).not_to have_content('Add a new attribute')
+      end
+    end
+
+    context 'with permissions' do
+      scenario 'remove attribute' do
+        login user
+
+        visit index_attribs_path(project: user.home_project_name)
+        click_link 'Delete attribute'
+        expect(find("#delete-attribute-modal-#{attribute.id}")).to have_text('Delete attribute?')
+        within("#delete-attribute-modal-#{attribute.id} .modal-footer") do
+          click_button('Delete')
+        end
+        sleep 1 # FIXME: Needed to avoid a flickering test.
```

Comment:
IIRC you could add a finder for the flash message (or any other element that would appear when the request was processed) here. That way we would use the timeout capybara uses.
I am not 100% this would work here, but I would give it a try.

Reply:
I'll try!

Reply:
@bgeuken I replaced `sleep 1` by 
```expect(find('#flash')).to have_text('Attribute sucessfully deleted!')```
But it also fails in CircleCI :disappointed:

Reply:
Any other idea?

Reply:
the problem is that find('#flash') is an instant return of nil and repeating the expect won't help. So you need expect something repeatable first.

`expect(page).to have_css('#flash');`
and *then* you can expect the flash to have text.